docs/content/docs/connectors/datastream/kinesis.md

```diff
@@ -27,644 +27,348 @@ under the License.
 
-consumerConfig.put(ConsumerConfigConstants.EFO_REGISTRATION_TYPE, 
-    ConsumerConfigConstants.EFORegistrationType.EAGER.name());
+### Watermark Handling in the source
 
-StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
-
-DataStream<String> kinesis = env.addSource(new FlinkKinesisConsumer<>(
-    "kinesis_stream_name", new SimpleStringSchema(), consumerConfig));
-```
-{{< /tab >}}
-{{< tab "Scala" >}}
-```scala
-val consumerConfig = new Properties()
-consumerConfig.put(AWSConfigConstants.AWS_REGION, "us-east-1")
-consumerConfig.put(ConsumerConfigConstants.STREAM_INITIAL_POSITION, "LATEST")
+The `KinesisStreamsSource` supplies the `approximateArrivalTimestamp` provided by Kinesis as the event time associated with each record read.
+For more information on event time handling in Flink, see [Event time]({{< ref "docs/concepts/time" >}}). 
+Note that this timestamp is typically referred to as a Kinesis server-side timestamp, and there are no guarantees 
+about the accuracy or order correctness (i.e., the timestamps may not always be ascending).
 
-consumerConfig.put(ConsumerConfigConstants.RECORD_PUBLISHER_TYPE, 
-    ConsumerConfigConstants.RecordPublisherType.EFO.name());
-consumerConfig.put(ConsumerConfigConstants.EFO_CONSUMER_NAME, "my-flink-efo-consumer");
+### Event Time Alignment for Shard Readers
 
-consumerConfig.put(ConsumerConfigConstants.EFO_REGISTRATION_TYPE, 
-    ConsumerConfigConstants.EFORegistrationType.EAGER.name());
+Since the shards are being consumed in parallel, some shards might be reading records far ahead from other shards.
+Flink supports synchronization between these reading speed using split-specific watermark alignment. The `KinesisStreamsSource`
+supports split-specific watermark alignment, and will pause reading from specific shards if the watermark from that shard
+is too far ahead of others. It will resume reading from that specific split once the other shards catch up.
 
-val env = StreamExecutionEnvironment.getExecutionEnvironment()
+### Threading Model
 
-val kinesis = env.addSource(new FlinkKinesisConsumer[String](
-    "kinesis_stream_name", new SimpleStringSchema, consumerConfig))
-```
-{{< /tab >}}
-{{< tab "Python" >}}
-```python
-consumer_config = {
-    'aws.region': 'us-east-1',
-    'flink.stream.initpos': 'LATEST',
-    'flink.stream.recordpublisher':  'EFO',
-    'flink.stream.efo.consumername': 'my-flink-efo-consumer',
-    'flink.stream.efo.registration': 'EAGER'
-}
+The `KinesisStreamsSource` uses multiple threads for shard discovery and data consumption.
 
-env = StreamExecutionEnvironment.get_execution_environment()
+#### Shard Discovery
 
-kinesis = env.add_source(FlinkKinesisConsumer(
-    "kinesis_stream_name", SimpleStringSchema(), consumer_config))
-```
-{{< /tab >}}
-{{< /tabs >}}
+For shard discovery, the `SplitEnumerator` runs on the JobManager to periodically discover new shards using the `ListShard` API.
```

Comment:
Link "periodically discover" to "Internally Used Kinesis APIs > Shard discovery"

Reply:
added